react/features/base/participants/middleware.js

```diff
@@ -67,10 +89,36 @@ MiddlewareRegistry.register(store => next => action => {
 
     case PARTICIPANT_JOINED:
     case PARTICIPANT_UPDATED: {
-        if (typeof APP !== 'undefined') {
-            const participant = action.participant;
-            const { id, local } = participant;
+        const { participant } = action;
+        const { id, local, raisedHand } = participant;
+
+        // Send an external update of the participant's raised hand state if a
+        // new raised hand state is defined the action.
+        if (typeof raisedHand !== 'undefined') {
+            if (local) {
+                conference.setLocalParticipantProperty(
+                    'raisedHand', raisedHand);
+            }
+
+
+            if (typeof APP === 'object' && local) {
+                APP.conference.isHandRaised = raisedHand;
+                APP.UI.onLocalRaiseHandChanged(raisedHand);
+                APP.UI.setLocalRaisedHandStatus(raisedHand);
+            } else if (typeof APP === 'object') {
+                const remoteParticipant
+                    = getParticipantById(store.getState(), id);
+
+                remoteParticipant
+                    && APP.UI.setRaisedHandStatus(
+                        remoteParticipant.id,
+                        remoteParticipant.name,
+                        raisedHand);
+            }
+        }
 
+        // Notify external listeners of potential avatarURL changes.
+        if (typeof APP === 'object') {
```

Comment:
Also not your code, but do you have an idea how we could get rid of this constant checking of APP without actually do the reac-reduxification if the old code? It really bothers me to see these all around the codebase.

Reply:
The web UI, and some of its dangling modules like the iframe API, has to be converted over to react/redux. The checking of APP is a way to to continue converting logic into redux without having to hook non-reactified web UI into redux.

Reply:
I just realized I gave you a non-answer. My answer is: just finally buckle down and convert it. It's probably a few months of work to do but it's where the codebase should get to anyway.

Reply:
This did get me thinking more...It could be possible to have a subscriber (or just another middleware) added to the redux store that pubsubs one thing out and then maybe on web some module is listening for events and then re-pub-subbing. Or even grouping all APP middleware into one place, which is similar. So I guess my solution just entails grouping all the APP checking middleware into one place, even if it means some duplicated logic. I still think committing to getting the old code converted is the better option for the long term, but honestly that long term might be forever.

Reply:
Yeah, you're probably right, especially if we already have it in the loop. We should then just push it to make it happen